scrapy/spidermiddlewares/base.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from scrapy import Request, Spider
+
+if TYPE_CHECKING:
+    from collections.abc import AsyncIterable, Iterable
+
+    from scrapy.http import Response
+
+
+class BaseSpiderMiddleware:
+    """Optional base class for spider middlewares.
+
+    This class provides helper methods for asynchronous ``process_spider_output``
+    methods. Middlewares that don't have a ``process_spider_output`` method don't need
+    to use it.
+    """
+
+    def process_spider_output(
+        self, response: Response, result: Iterable[Any], spider: Spider
+    ) -> Iterable[Any]:
+        for o in result:
+            if isinstance(o, Request):
+                o = self._process_request(o, response, spider)
+                if o is not None:
+                    yield o
```

Comment:
I wonder if we should use `yield from self.process_request(o, response)`, to allow yielding multiple items/requests based on a given request (or item below), and to keep `spider` off since it feels like an obsolete parameter not needed nowadays with the 1 crawler = 1 spider constraint.

Reply:
`yield from` would require users to yield the items, instead of returning them, right?

Reply:
Yes.